catalog/transformer/catalog-transformer-mgmp/src/main/java/org/codice/alliance/catalog/transformer/mgmp/MgmpTransformer.java

```diff
@@ -0,0 +1,604 @@
+                MgmpConstants.MGMP_SPATIAL_REFERENCE_SYSTEM_PROJECTION_NUMBER_TYPE_PATH,
+                false);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                mgmpTypes,
+                MgmpConstants.MGMP_SPATIAL_REFERENCE_SYSTEM_GRID_NUMBER_TYPE_PATH,
+                false);
+
+        ArrayList<String> mgmpCodes = new ArrayList<>();
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                mgmpCodes,
+                MgmpConstants.MGMP_SPATIAL_REFERENCE_SYSTEM_TYPE_PATH,
+                false);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                mgmpCodes,
+                MgmpConstants.MGMP_SPATIAL_REFERENCE_SYSTEM_DATUM_TYPE_PATH,
+                false);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                mgmpCodes,
+                MgmpConstants.MGMP_SPATIAL_REFERENCE_SYSTEM_ELLIPSOID_TYPE_PATH,
+                false);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                mgmpCodes,
+                MgmpConstants.MGMP_SPATIAL_REFERENCE_SYSTEM_PROJECTION_TYPE_PATH,
+                false);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                mgmpCodes,
+                MgmpConstants.MGMP_SPATIAL_REFERENCE_SYSTEM_GRID_TYPE_PATH,
+                false);
+
+        List<String> crsCodes = new ArrayList<>();
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                crsCodes,
+                MgmpConstants.MGMP_SPATIAL_REFERENCE_SYSTEM_CRS_WKT_TYPE_PATH,
+                true);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                crsCodes,
+                MgmpConstants.MGMP_SPATIAL_REFERENCE_SYSTEM_GRID_WKT_TYPE_PATH,
+                true);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                crsCodes,
+                MgmpConstants.MGMP_SPATIAL_REFERENCE_SYSTEM_PROJECTION_WKT_TYPE_PATH,
+                true);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                crsCodes,
+                MgmpConstants.MGMP_SPATIAL_REFERENCE_SYSTEM_ELLIPSOID_WKT_TYPE_PATH,
+                true);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                crsCodes,
+                MgmpConstants.MGMP_SPATIAL_REFERENCE_SYSTEM_DATUM_WKT_TYPE_PATH,
+                true);
+
+        List<String> codes = pathValueTracker.getAllValues(toPath(GmdConstants.CRS_CODE_PATH));
+        List<String> types = pathValueTracker.getAllValues(toPath(GmdConstants.CRS_AUTHORITY_PATH));
+        if (CollectionUtils.isNotEmpty(types) && CollectionUtils.isNotEmpty(codes)) {
+            for (int i = 0; i < types.size() && i < codes.size(); i++) {
+                crsCodes.add(types.get(i) + ":" + codes.get(i));
+            }
+        }
+
+        if (CollectionUtils.isNotEmpty(mgmpTypes) && CollectionUtils.isNotEmpty(mgmpCodes)) {
+            for (int i = 0; i < mgmpCodes.size() && i < mgmpTypes.size(); i++) {
```

Comment:
Also, while incredibly unimportant, it bothers me a little that this loop does not test the size of the collections in the same order their elements will be concatenated together.